src/api/client-event-manager-api.js

```diff
@@ -72,7 +72,8 @@ export class ClientEventManagerApi {
    * stop the event if a filterer cancels it.  After that, it will call each
    * listener asynchronously.
    * @param {!ClientEvent} event
+   * @param {Object} eventParams
```

Comment:
I believe if we make it optional, it's allowed to be null/undefined, but the compiler will still complain if the number of arguments don't match. I can still change it to be optional, but I think it is more clear to the callers if it is passing an empty object rather than null..

Reply:
Optional and nullable are different. I think the right syntax here is `@param {!Object=} eventParams` to make it optional (maybe undefined but if present, must not be null).
https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System#nullable-vs-optional-parameters-and-properties

Reply:
I see, allowing undefined makes the function overloading work!

Reply:
This still doesn't get around the issue for me for when we call `logEvent` (above comment) or when we register listeners / bind the callback function:
```
src/runtime/client-event-manager.js:109:25: ERROR - [JSC_TYPE_MISMATCH] actual parameter 1 of Array.prototype.push does not match formal parameter
found 
```
```
src/runtime/google-analytics-event-listener.js:37:6: ERROR - [JSC_TYPE_MISMATCH] actual parameter 1 of ClientEventManager$$module$src$runtime$client_event_manager.prototype.registerEventListener does not match formal parameter
found 
```

Reply:
seems like I was able to resolve this with some additional `|undefined` annotations, marking resolved.